ipaserver/install/httpinstance.py

```diff
@@ -237,63 +191,53 @@ def configure_gssproxy(self):
         tasks.configure_http_gssproxy_conf(IPAAPI_USER)
         services.knownservices.gssproxy.restart()
 
-    def change_mod_nss_port_from_http(self):
-        # mod_ssl enforces SSLEngine on for vhost on 443 even though
-        # the listener is mod_nss. This then crashes the httpd as mod_nss
-        # listened port obviously does not match mod_ssl requirements.
-        #
-        # The workaround for this was to change port to http. It is no longer
-        # necessary, as mod_nss now ships with default configuration which
-        # sets SSLEngine off when mod_ssl is installed.
-        #
-        # Remove the workaround.
-        if sysupgrade.get_upgrade_state('nss.conf', 'listen_port_updated'):
-            installutils.set_directive(paths.HTTPD_NSS_CONF, 'Listen', '443', quotes=False)
-            sysupgrade.set_upgrade_state('nss.conf', 'listen_port_updated', False)
-
-    def __set_mod_nss_port(self):
-        self.fstore.backup_file(paths.HTTPD_NSS_CONF)
-        if installutils.update_file(paths.HTTPD_NSS_CONF, '8443', '443') != 0:
-            print("Updating port in %s failed." % paths.HTTPD_NSS_CONF)
-
-    def __set_mod_nss_nickname(self, nickname):
-        quoted_nickname = installutils.quote_directive_value(
-            nickname, quote_char="'")
-        installutils.set_directive(
-            paths.HTTPD_NSS_CONF, 'NSSNickname', quoted_nickname, quotes=False)
-
     def get_mod_nss_nickname(self):
         cert = installutils.get_directive(paths.HTTPD_NSS_CONF, 'NSSNickname')
         nickname = installutils.unquote_directive_value(cert, quote_char="'")
         return nickname
 
-    def set_mod_nss_protocol(self):
-        installutils.set_directive(paths.HTTPD_NSS_CONF, 'NSSProtocol', 'TLSv1.0,TLSv1.1,TLSv1.2', False)
-
-    def enable_mod_nss_renegotiate(self):
-        installutils.set_directive(paths.HTTPD_NSS_CONF, 'NSSRenegotiation', 'on', False)
-        installutils.set_directive(paths.HTTPD_NSS_CONF, 'NSSRequireSafeNegotiation', 'on', False)
-
-    def disable_mod_nss_ocsp(self):
-        if sysupgrade.get_upgrade_state('http', NSS_OCSP_ENABLED) is None:
-            self.__disable_mod_nss_ocsp()
-            sysupgrade.set_upgrade_state('http', NSS_OCSP_ENABLED, False)
-
-    def __disable_mod_nss_ocsp(self):
+    def backup_ssl_conf(self):
+        self.fstore.backup_file(paths.HTTPD_SSL_CONF)
+
+    def disable_nss_conf(self):
+        # There is no safe way to co-exist since there is no safe port
+        # to make mod_nss use, disable it completely.
+        if os.path.exists(paths.HTTPD_NSS_CONF):
+            installutils.remove_file(paths.HTTPD_NSS_CONF)
```

Comment:
You can’t just remove a file we don’t own. If the user decides to uninstall IPA, his system may be broken should they try to install something relying on mod_nss, whichever devil that is. Removing the file also breaks the scenario when you uninstall IPA and then try to downgrade the packages and install it again.

Reply:
It should be restored on uninstallation.

The problem is that there is a listen port defined in nss.conf and even if the engine is off Apache is going to listen on that port. So what port should be used? I punted.

What we could do instead is to move nss.conf to nss.conf.ipa in the same directory and create a zero-length nss.conf. If an admin wanted to restore it they could and we would let them deal with the port conflict. It just makes uninstallation quite a bit more complex.

Reply:
My point was exactly as you're saying - it should be restore on uninstallation. You can just push the file to `fstore` that the HTTPInstance uses and restore it at the point where all other files are being restored.

Reply:
I think the scenario I was thinking of was: you have IPA installed with mod_nss, you upgrade and things get converted to mod_ssl and mod_nss is backed up. You uninstall IPA. The restored mod_nss still has IPA configured.